scripts/list_hwm.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+from pathlib import Path, PurePath
+import pykwalify.core
+import sys
+import yaml
+
+SOC_METADATA_SCHEMA = '''
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The socs.yml file is a simple list of key value pairs containing SoCs
+# and their location which is used by the build system.
+type: map
+mapping:
+  socs:
+    required: true
+    type: seq
+    sequence:
+      - type: map
+        mapping:
+          name:
+            required: false
+            type: str
+            desc: Name of the SoC
```

Comment:
Any reason we wouldn't want the name of a SoC to be required?

Reply:
The important piece of information is to know the location of Kconfig and CMake files for the build system.

Initially the field was intended to be of informational use, and I wanted to avoid an ever growing list.
For example, if a vendor uses a common SoC family folder, they can just add new SoCs without having to keep updating the yaml list.
It is enough to just specify where to find a given family.

Similar if a SoC is part of a series, then it's possible to specify just where the series exists, not having to identify each individual SoC available in the yaml file.

Therefore the name is optional.

Note, this comment has made me identify a potential double sourcing issue in CMake in case both a soc name and series / family is specified. Looking into fixing that.